tests/functional/adapter/test_override_database/test_override_database.py

```diff
@@ -0,0 +1,158 @@
+            "vars": {
+                "alternate_db": os.getenv("BIGQUERY_TEST_ALT_DATABASE"),
+            },
+            "quoting": {
+                "database": True,
+            },
+            "seeds": {
+                "quote_columns": False,
+            }
+        }
+
+
+class TestModelOverrideBigQuery(BaseOverrideDatabase):
+    def run_database_override(self, project):
+        run_dbt(["seed"])
+        assert len(run_dbt(["run"])) == 4
+        check_relations_equal_with_relations(project.adapter, [
+            project.adapter.Relation.create(database=os.getenv("BIGQUERY_TEST_DATABASE"), schema=project.test_schema, identifier="seed"),
+            project.adapter.Relation.create(database=os.getenv("BIGQUERY_TEST_ALT_DATABASE"), schema=project.test_schema, identifier="view_2"),
+            project.adapter.Relation.create(database=os.getenv("BIGQUERY_TEST_DATABASE"), schema=project.test_schema, identifier="view_1"),
+            project.adapter.Relation.create(database=os.getenv("BIGQUERY_TEST_DATABASE"), schema=project.test_schema, identifier="view_3"),
+            project.adapter.Relation.create(database=os.getenv("BIGQUERY_TEST_ALT_DATABASE"), schema=project.test_schema, identifier="view_4")
+        ])
+
+
+    def test_bigquery_database_override(self, project):
+        self.run_database_override(project)
+
+
+class BaseTestProjectModelOverrideBigQuery(BaseOverrideDatabase):
+
+    def run_database_override(self, project):
+        run_dbt(["seed"])
+        assert len(run_dbt(["run"])) == 4
+        self.assertExpectedRelations(project)
+
+    def assertExpectedRelations(self, project):
+        check_relations_equal_with_relations(project.adapter, [
+            project.adapter.Relation.create(database=os.getenv("BIGQUERY_TEST_DATABASE"), schema=project.test_schema, identifier="seed"),
+            project.adapter.Relation.create(database=os.getenv("BIGQUERY_TEST_ALT_DATABASE"), schema=project.test_schema, identifier="view_2"),
+            project.adapter.Relation.create(database=os.getenv("BIGQUERY_TEST_ALT_DATABASE"), schema=project.test_schema, identifier="view_1"),
+            project.adapter.Relation.create(database=os.getenv("BIGQUERY_TEST_DATABASE"), schema=project.test_schema, identifier="view_3"),
+            project.adapter.Relation.create(database=os.getenv("BIGQUERY_TEST_ALT_DATABASE"), schema=project.test_schema, identifier="view_4")
+        ])
+
+
+class TestProjectModelOverrideBigQuery(BaseTestProjectModelOverrideBigQuery):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "vars": {
+                "alternate_db": os.getenv("BIGQUERY_TEST_ALT_DATABASE"),
+            },
+            "models": {
+                "database": os.getenv("BIGQUERY_TEST_ALT_DATABASE"),
+                "test": {
+                    "subfolder": {
+                        "database": os.getenv("BIGQUERY_TEST_DATABASE")
```

Comment:
You could replace this env var pretty straightforwardly with either of the following.

More like what users would actually write in their projects:
```suggestion
                        "database": "{{ target.database }}"
```

More like what this is doing currently:
```suggestion
                        "database": project.database
```

Reply:
I think the target.database works more as that part of the function doesn't generally have access to project and when i tried to tap it it gave recursive dependency issues.